packages/client/src/StreamrClient.ts

```diff
@@ -93,6 +96,22 @@ class StreamrClientBase implements Context {
 
         this.onDestroy = this.destroySignal.onDestroy.bind(this.destroySignal)
         this.isDestroyed = this.destroySignal.isDestroyed.bind(this.destroySignal)
+
+        metricsPublisher.init()
+    }
+
+    /**
+     * @category Important
+     */
+    async publish<T>(
+        streamDefinition: StreamDefinition,
+        content: T,
+        timestamp: string | number | Date = Date.now(),
+        partitionKey?: string | number
+    ): Promise<StreamMessage<T>> {
+        const result = await this.publisher.publish(streamDefinition, content, timestamp, partitionKey)
+        this.eventEmitter.emit('publish', undefined)
```

Comment:
Not possible to simplify?
```suggestion
        this.eventEmitter.emit('publish')
```

Reply:
Necessary to compliation